chia/wallet/singleton_wallet.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from typing import List, Tuple
+
+from chia_rs import CoinSpend
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint32
+from chia.wallet.transaction_record import TransactionRecord
+from chia.wallet.wallet import Wallet
+from chia.wallet.wallet_protocol import WalletProtocol
+
+
+class SingletonWallet(WalletProtocol):
+    """
+
+    Event:
+    A singleton eve coin is detected by the WalletStateManager.
+    The WSM does not have a wallet for that launcher_id
+    - create_wallet_from_coin is called
+    - wallet created in DB
+    -
+
+
+    Event:
+    A singleton spend is detected for an existing (WSM detects existing wallet by ____
+    - apply_state_transition
+
+
+    Issues:
+        These names ... could be better
+        What about multiple Singletons being managed from the same wallet?
+        What about coins not following the singleton pattern?
+
+
+
+    Generalized inner/outer wallet ideas:
+    - detect outer (CAT / Singleton)
+    - detect inner (does this coin belong to me)
+    - create from spend
+    - create new wallet for singleton that already exists (we have not yet received the coin)
+    - spending when you don't own the singleton, but you can spend
+      - Examples:
+    """
+
+    ###################### Class methods ######################################
+
+    # Name: detect_singleton_coin_type ?
+    @classmethod
+    def coin_is_my_singleton_wallet_type(cls, coin: Coin, inner_puzzle: Program) -> bool:
+        """Match the singleton inner puzzle for this type of wallet"""
+        ...
+
+    # def register_singleton_inner_puzzle(self) -> Program:
+    #    ...
+
+    @classmethod
+    def create_wallet_from_coin(
```

Comment:
Do we want the in-memory wallet : in-db state table relationship to be 1:1 ?

Dow we want to allow one `Wallet` object to handle multiple layered coins in the simple interface?